iceoryx_utils/include/iceoryx_utils/internal/cxx/forward_list.inl

```diff
@@ -0,0 +1,581 @@
+    return ++citer;
+}
+
+
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::iterator forward_list<T, CAPACITY>::end() noexcept
+{
+    return iterator{this, INVALID_INDEX};
+}
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::const_iterator forward_list<T, CAPACITY>::end() const noexcept
+{
+    return cend();
+}
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::const_iterator forward_list<T, CAPACITY>::cend() const noexcept
+{
+    return const_iterator{this, INVALID_INDEX};
+}
+
+
+template <typename T, int64_t CAPACITY>
+inline bool forward_list<T, CAPACITY>::empty() const noexcept
+{
+    return (m_size == 0);
+}
+
+template <typename T, int64_t CAPACITY>
+inline bool forward_list<T, CAPACITY>::full() const noexcept
+{
+    return (m_size >= CAPACITY);
+}
+
+template <typename T, int64_t CAPACITY>
+inline typename forward_list<T, CAPACITY>::size_type forward_list<T, CAPACITY>::size() const noexcept
+{
+    return m_size;
+}
+
+template <typename T, int64_t CAPACITY>
+inline typename forward_list<T, CAPACITY>::size_type forward_list<T, CAPACITY>::capacity() const noexcept
+{
+    return CAPACITY;
+}
+
+template <typename T, int64_t CAPACITY>
+inline typename forward_list<T, CAPACITY>::size_type forward_list<T, CAPACITY>::max_size() const noexcept
+{
+    return capacity();
+}
+
+
+template <typename T, int64_t CAPACITY>
+template <typename... ConstructorArgs>
+bool forward_list<T, CAPACITY>::emplace_front(ConstructorArgs&&... args) noexcept
+{
+    return (cend() != emplace_after(before_begin(), std::forward<ConstructorArgs>(args)...));
+}
+
+template <typename T, int64_t CAPACITY>
+template <typename... ConstructorArgs>
+typename forward_list<T, CAPACITY>::iterator
+forward_list<T, CAPACITY>::emplace_after(const_iterator afterToBeEmplacedIter, ConstructorArgs&&... args) noexcept
+{
+    if (this != afterToBeEmplacedIter.m_cList)
+    {
+        errorMessage(__PRETTY_FUNCTION__, " iterator of other list can't be used ");
+        std::terminate();
+    }
+
+    const size_type toBeAddedIdx = getHeadOfFreeElementList()->nextIdx;
+
+    if (m_size < CAPACITY && toBeAddedIdx < INVALID_INDEX)
+    {
+        auto newNodePointer = getNodePtrFromIdx(toBeAddedIdx);
+
+        // unlink from freeList
+        getHeadOfFreeElementList()->nextIdx = newNodePointer->nextIdx;
+
+        // data class c'tor
+        new (&(newNodePointer->data)) T(std::forward<ConstructorArgs>(args)...);
+
+        // add to usedList
+        const size_type afterToBeEmplacedIdx = afterToBeEmplacedIter.m_iterListNodeIdx;
+        const size_type emplaceBeforeIdx = getNodePtrFromIdx(afterToBeEmplacedIdx)->nextIdx;
+        newNodePointer->nextIdx = emplaceBeforeIdx;
+        getNodePtrFromIdx(afterToBeEmplacedIdx)->nextIdx = toBeAddedIdx;
+
+        ++m_size;
+
+        return iterator{const_cast<forward_list<T, CAPACITY>*>(afterToBeEmplacedIter.m_cList), toBeAddedIdx};
```

Comment:
const_cast only required because m_clist is const, I think it should not be const for this reason. I placed a comment were it is defined.